src/test/java/co/rsk/federate/signing/hsm/config/PowHSMConfigTest.java

```diff
@@ -0,0 +1,171 @@
+package co.rsk.federate.signing.hsm.config;
+
+import static co.rsk.federate.signing.hsm.config.PowHSMConfigParameter.*;
+import static co.rsk.federate.signing.utils.TestUtils.createHash;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.federate.signing.config.SignerConfig;
+import co.rsk.federate.signing.hsm.HSMBlockchainBookkeepingRelatedException;
+import co.rsk.federate.signing.hsm.HSMClientException;
+import co.rsk.federate.signing.hsm.HSMUnsupportedTypeException;
+import co.rsk.federate.signing.hsm.client.HSMBookkeepingClient;
+import co.rsk.federate.signing.hsm.message.PowHSMBlockchainParameters;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import java.math.BigInteger;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class PowHSMConfigTest {
+
+  private PowHSMConfig powHsmConfig;
+  private final SignerConfig signerConfig = mock(SignerConfig.class);
+  private final Config config = mock(Config.class);
+  private final HSMBookkeepingClient hsmClient = mock(HSMBookkeepingClient.class);
+
+  @BeforeEach
+  void setup() throws Exception {
+    when(signerConfig.getConfig()).thenReturn(config);
+    when(signerConfig.getType()).thenReturn("hsm");
+    powHsmConfig = new PowHSMConfig(signerConfig);
+  }
+
+  @Test
+  void getDifficultyTarget_whenCallToPowHSMBlockchainParametersSucceeds_shouldRetriveConfigValueFromPowHSM()
+      throws Exception {
+    BigInteger expectedDifficultyTarget = BigInteger.valueOf(1L);
+    PowHSMBlockchainParameters response = new PowHSMBlockchainParameters(
```

Comment:
I'd put the first parameter in a variable, reading this it's hard to know what it represents

Reply:
Updated, created var `blockCheckpoint`